### PR TITLE
Add `.coderabbit.yml` config and update pull request template

### PR DESCRIPTION
**Pull request by seanbudd**
## Summary by CodeRabbit

- **New Features**
  - Introduced CodeRabbit integration with customizable review settings and tools.
  - Added an AI high-level summary section to the pull request template to streamline PR documentation.

### Link to issue number:
None

### Summary of the issue:
[CodeRabbit](https://coderabbit.ai/) is a tool NV Access has been trialling for code reviews in NVDA.
AI code reviews create helpful summaries of PRs, and pick up issues other reviews may miss, or pick up issues before another reviewer gets to it, saving time.
Now that we are somewhat happy with the current config, we should commit it to the repository rather than rely on the GUI.
This means contributors can more easily improve the auto review process such as improving the path prompts and adding learnings.

### Description of user facing changes
AI code reviews will now happen automatically.

### Description of development approach
Followed steps in https://docs.coderabbit.ai/guides/configure-coderabbit/

### Testing strategy:
Will be tested over time, improving prompts, etc

### Known issues with pull request:
None

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

File: .coderabbit.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    auto_incremental_review: false
+    base_branches:
+      - master
+      - beta
+      - rc
+  path_instructions:
+    - path: "**/*"
+      instructions: "
+      Focus on code smells, logic errors, edge cases, missing test cases, security flaws and serious issues.
+      Avoid commenting on minor issues such as linting, formatting and style issues.
```
Comment (LeonarddeR) [inline .coderabbit.yml:25]: Some suggestions, particularly regarding handling of documentation.
```suggestion
      Avoid commenting on minor issues such as linting, formatting and style issues.
      Check spelling of changes in markdown (*.md)files and suggest improvements in spelling/grammar where appropriate.
      Ensure that changes in markdown files (*.md) follow the style of one sentence per line, suggest changes if this is not met.
```

Comment (seanbudd) [reply .coderabbit.yml:25]: There's a default setting currently enabled for spell checking, with much more complex configuration available, but the sentence per line mention is worth adding

Comment (seanbudd) [reply .coderabbit.yml:25]: I've fixed this up now

**Review by gerald-hartig — APPROVED** [review]
